### DISCUSSION TITLE
Incremental backups with snapshotting possible? If yes, how?

### DISCUSSION THREAD

**Discussion by schoenfbe** [minio/minio#15562] [Q&A] [answered]
Hi, I'm kinda new to MinIO and struggling a bit with backups.
We've got two minio distributions separately, which both should be backed up also separately to backup MinIOs.
I implemented it that it mirrors the original to the backup minio, and so far, that worked fine.
But now I'm getting a lot of errors like:
mc: <ERROR> Failed to perform mirroring, with error condition (mm-source-mtime) Overwrite not allowed for `randomfile.jpg`. Use `--overwrite` to override this behavior.
which seems to appear only for png/jpg/pdf files, text and code files seem to be fine and mirrored still after changes.
My understanding was that mirror incrementally changes the files, so that when I have a file on the source, it gets duplicated on the target, and if I change the sourcefile, the target also changes.
This, however, doesn't seem to happen as per that failure notice, and also the docu for --overwrite states: "Without --overwrite, if an object already exists on the Destination, the mirror process fails to synchronize that object. mc mirror logs an error and continues to synchronize other objects."
but I don´t want to overwrite the files, I just want to apply the changes, shouldn't mirror do exactly that?
the current command is simply "mc mirror old_minio new_minio"
Furthermore, if I can get that to work as intended, we'd like to have git-like snapshots/versions of files, so that we can look what has changed between two backed up versions, is that possible with minio?
Thanks for any help :)

**Comment by harshavardhana** [comment, marked as answer]
> mc: Failed to perform mirroring, with error condition (mm-source-mtime) Overwrite not allowed for randomfile.jpg. Use --overwrite to override this behavior.

Just run with `mc mirror -a`

**Reply by schoenfbe** [reply]
> > mc: Failed to perform mirroring, with error condition (mm-source-mtime) Overwrite not allowed for randomfile.jpg. Use --overwrite to override this behavior.
> 
> Just run with `mc mirror -a` 

Hi, thanks for your answer, but does that do what I want?
-a says "preserve bucket policy rules on target bucket(s)"
What exactly would that do compared to normal mc mirror?
Are changes within the same file mirrored with that command, as mirror itself doesn't seem to do that?
Thanks :)

**Reply by schoenfbe** [reply]
What I would ultimately be after would be a git-like versioning, so we could look at exactly what had changed between file versions

**Reply by schoenfbe** [reply]
nevermind, found out that I can achieve what I want with versioning even on a single drive, just took a while to find it, and then the --overwrite flag does exactly what I want
